include/hip/hcc_detail/math_functions.h

```diff
@@ -119,6 +120,43 @@ uint64_t __make_mantissa(const char* tagp)
     return __make_mantissa_base10(tagp);
 }
 
+// DOT FUNCTIONS
+__DEVICE__
+inline
+float amd_mixed_dot(__2f16 a, __2f16 b, float c, bool saturate) {
```

Comment:
I'm not sure this is the best argument type.  Developers will probably want to use half2.  This function may need to bitcast a and b to the ockl types.

Reply:
@AlexVlx I noticed that half2 is being defined in hip_fp16.h as a struct. Do we want to change half2 into ockl type half2? or keep it as is? What are your thoughts

Reply:
I think the types of the and and b arguments should be whatever developers are likely to use, and that we should translate (i.e. bitcast) it to the type ockl expects in this function.  I think they are more likely to use half2 than __2f16.

Reply:
@aaronenyeshi You absolutely do not want to touch the types that are part of the HIP public interface, which must match CUDA behaviourally. This is part of HIP's contract, those are law, you do not want to turn them into ockl anything. You want the AMD specific stuff to interoperate seamlessly, so you want the public interface to take HIP types. `__half2_raw::data` gives you exactly what you want.